### DISCUSSION TITLE
Please help Inteli7-8700 CPU PERFORMANCE OPTIMIZE

### DISCUSSION THREAD

**Discussion by jackgump** [madMAx43v3r/chia-plotter#804] [Q&A] [answered]
ubuntu20.04.2 LTS .
Intel® Core™ i7-8700 CPU @ 3.20GHz × 12.
nvme0   storage   Samsung SSD 980 1TB, partition type:  NTFS/exFAT/HPFS (Bootable).
memory  16GiB,   USCORSAIR DDR4 3600
HD  8T,  Seagate ST8000NM000A, partition type:Linux Filesystem.

Total plot creation time was 9965.78 sec (166.096 min), for the following plotting setting, is it normal speed??

jack@jack-chia1:~/chia-plotter/build$ ./chia_plot -n 1 -r 20  --tmpdir /media/jack/数据/temp1/ --tmpdir2 /media/jack/数据/temp2/ --finaldir /media/jack/8t/ -c xch1ufy36p86jk9cr4nlyqx5z3zvzmm0ccf4us09pjtkp06r3ztj2vhqewx6f3 -f ad49fe2c8b76ee1d300a01b0502184b3e17b6522fd84ff49b3711e60c1afe31666a008f85062b7358d101d09a7323059
Multi-threaded pipelined Chia k32 plotter - 95389dc
(Sponsored by Flexpool.io - Check them out if you're looking for a secure and scalable Chia pool)

Final Directory: /media/jack/8t/
Number of Plots: 1
Crafting plot 1 out of 1
Process ID: 17058
Number of Threads: 20
Number of Buckets P1:    2^8 (256)
Number of Buckets P3+P4: 2^8 (256)
Pool Puzzle Hash:  e2491d04fa958b81d67f200d41444c16f6fc6135e41e50c9760bf4388972532e
Farmer Public Key: ad49fe2c8b76ee1d300a01b0502184b3e17b6522fd84ff49b3711e60c1afe31666a008f85062b7358d101d09a7323059
Working Directory:   /media/jack/数据/temp1/
Working Directory 2: /media/jack/数据/temp2/
Plot Name: plot-k32-2021-07-12-15-31-e831c671c299be0ccf82882ead36aac21df6e67a5ef3172d0f1138db9089ed6d
[P1] Table 1 took 153.477 sec
[P1] Table 2 took 530.619 sec, found 4295102891 matches
[P1] Table 3 took 698.879 sec, found 4295129831 matches
[P1] Table 4 took 758.911 sec, found 4295031636 matches
[P1] Table 5 took 689.224 sec, found 4295192131 matches
[P1] Table 6 took 613.878 sec, found 4295133864 matches
[P1] Table 7 took 520.949 sec, found 4295172522 matches
Phase 1 took 3966.16 sec
[P2] max_table_size = 4295192131
[P2] Table 7 scan took 77.5417 sec
[P2] Table 7 rewrite took 289.532 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 97.517 sec
[P2] Table 6 rewrite took 234.79 sec, dropped 581343830 entries (13.5349 %)
[P2] Table 5 scan took 93.2762 sec
[P2] Table 5 rewrite took 225.767 sec, dropped 762134403 entries (17.7439 %)
[P2] Table 4 scan took 95.1484 sec
[P2] Table 4 rewrite took 252.523 sec, dropped 828872796 entries (19.2984 %)
[P2] Table 3 scan took 91.7016 sec
[P2] Table 3 rewrite took 251.688 sec, dropped 855220410 entries (19.9114 %)
[P2] Table 2 scan took 92.001 sec
[P2] Table 2 rewrite took 283.773 sec, dropped 865675974 entries (20.155 %)
Phase 2 took 2090.44 sec
Wrote plot header with 252 bytes
[P3-1] Table 2 took 404.251 sec, wrote 3429426917 right entries
[P3-2] Table 2 took 290.873 sec, wrote 3429426917 left entries, 3429426917 final
[P3-1] Table 3 took 250.584 sec, wrote 3439909421 right entries
[P3-2] Table 3 took 280.483 sec, wrote 3439909421 left entries, 3439909421 final
[P3-1] Table 4 took 249.058 sec, wrote 3466158840 right entries
[P3-2] Table 4 took 290.931 sec, wrote 3466158840 left entries, 3466158840 final
[P3-1] Table 5 took 259.197 sec, wrote 3533057728 right entries
[P3-2] Table 5 took 293.974 sec, wrote 3533057728 left entries, 3533057728 final
[P3-1] Table 6 took 264.982 sec, wrote 3713790034 right entries
[P3-2] Table 6 took 306.051 sec, wrote 3713790034 left entries, 3713790034 final
[P3-1] Table 7 took 349.654 sec, wrote 4295172522 right entries
[P3-2] Table 7 took 359.991 sec, wrote 4294967296 left entries, 4294967296 final
Phase 3 took 3601.19 sec, wrote 21877310236 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 307.919 sec, final plot size is 108836175416 bytes
Total plot creation time was 9965.78 sec (166.096 min)
Started copy to /media/jack/8t/plot-k32-2021-07-12-15-31-e831c671c299be0ccf82882ead36aac21df6e67a5ef3172d0f1138db9089ed6d.plot
Copy to /media/jack/8t/plot-k32-2021-07-12-15-31-e831c671c299be0ccf82882ead36aac21df6e67a5ef3172d0f1138db9089ed6d.plot finished, took 406.036 sec, 255.628 MB/s avg.
j

**Comment by maxpierpont** [comment, marked as answer]
just check resource usage while plotting, your system memory usage can be more than 16gb, and system starts swapping to the same disk you are plotting, bad idea plotting to system drive with low memory and swap enabled (try using other value for threads (-r) and buckets (-u) to lower memory usage; average 0,5gb of memory used per thread at 512buckets, 1gb per thread at 256buckets, so you should better use "-r 12" and lower for your cpu and 16gb if you use 256 buckets by default)
samsung 980 can be a bottleneck too:  its announced speed of up to 7000mb/s is only while it's plugged to nvme4.0 and writing to slc cache (114gb announced and lasts up to 170gb), and when cache gets full, drive speed goes very low until cache is flushed (but it's not flushed until you stop writing to it).

**Reply by jackgump** [reply]
Thank you so much!
